tests/serverpod_test_server/test_integration/test_tools/transaction_correctness_test.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+import 'serverpod_test_tools.dart';
+
+void main() {
+  withServerpod(
+    'Given transaction call in test',
+    (sessionBuilder, endpoints) {
+      var session = sessionBuilder.build();
+
+      test(
```

Comment:
Suggestion: Please add a comment that explains how this test would fail if we had not applied our current fix.

since this is a nitty implementation detail I think it makes sense to help future developers out a bit.

Reply:
b4fdc1a